### ISSUE TITLE
For something about the loss "nan"

### ISSUE THREAD

**Issue by stonewalking** [ilucasgoncalves/VAVL#10]
Hi author,

Thanks for the supports.

One thing i found when training a subset, there are quite a lot nan prints in interations(all labels are nan). Use pytorch abnomal dectect, it locals the total_loss.backward() actually meets some 0s and resulting nans.

While i assume it maybe about the autocast(), i found that you didn't use the gradscaler actually. After i add it , the train_dom still happens nan but the rest label are okay, although the lr is quite small 0.1e-3.

This things happens after i use the set_determinstic(seed) function in your code(）and then "#" ut. (After i  "#" the set_determinstic(seed) function.) 

Did you meet the same things? while using some other csv label file is fine, but the first the 3rd and 5rd happens like this. Is it about the some issues after you use set_determinstic(seed) function, the cuda env or gpu env changed somethings?

**Comment by ilucasgoncalves** [comment]
I remember having similar issues with "set_deterministic" in the past and the issue was due to some versions mismatch I had. Although, I did not come across this same issue  for this work. 

I tried reproducing the error and trying to perform anomaly detection with pytorch abnormal. But the same issue did not show up for me. I ran this testing on a GeForce RTX 3090, CUDA release 11.0, V11.0.194, and pytorch 1.12.1.
